apps/tangle-dapp/components/BnChartTooltip.tsx

```diff
@@ -0,0 +1,45 @@
+import { Typography } from '@webb-tools/webb-ui-components/typography/Typography';
+import BN from 'bn.js';
+import { TooltipProps } from 'recharts';
+
+import { ServiceType } from '../types';
+import { formatTokenBalance } from '../utils/polkadot/tokens';
+
+const BnChartTooltip = (
+  allocations: Record<ServiceType, BN | null>,
+  freeBalance: BN,
+  restakedAmount: BN
+) => {
+  const composition = ({
+    active,
+    payload,
+  }: TooltipProps<number, ServiceType>) => {
+    const role = payload?.[0]?.name;
```

Comment:
I think we need to add `useMemo` in all variables here because the value for `active` will be updated frequently

Reply:
Since the operations (ie. object access) are straightforward, using `useMemo` would not provide any benefit here since it is not an expensive operation that needs to be cached.